redis/src/types.rs

```diff
@@ -1211,8 +1212,7 @@ impl FromRedisValue for String {
 impl<T: FromRedisValue> FromRedisValue for Vec<T> {
     fn from_redis_value(v: &Value) -> RedisResult<Vec<T>> {
         match *v {
-            // this hack allows us to specialize Vec<u8> to work with
-            // binary data whereas all others will fail with an error.
+            // All binary data except u8 will try to parse into a single element vector.
             Value::Data(ref bytes) => match FromRedisValue::from_byte_vec(bytes) {
                 Some(x) => Ok(x),
                 None => invalid_type_error!(v, "Response type not vector compatible."),
```

Comment:
I think the message `"Response type not vector compatible."` is no longer appropriate here, as any failure here will now be due to the failure in calling `Self::from_redis_value`

Reply:
Agree! How about `Fail to convert strint-data("any") via Self::from_redis_value`?

Reply:
As long as it's clear and spelled correctly, it's fine. I think something as simple as this would probably be ok:

```None => invalid_type_error!(v, "Conversion to Vec<T> failed."),```

Reply:
Sure, I think that makes sense!

Reply:
Updated! Thanks for the suggestion!